python/taichi/linalg/sparse_matrix.py

```diff
@@ -198,6 +198,50 @@ def build_from_ndarray(self, ndarray):
                 'Sparse matrix only supports building from [ti.ndarray, ti.Vector.ndarray, ti.Matrix.ndarray]'
             )
 
+    def build_csr_cusparse(self, data, indices, indptr):
+        """Build a csr format sparse matrix using cuSparse where the column indices
+            for row i are stored in ``indices[indptr[i]:indptr[i+1]]``
+            and their corresponding values are stored in ``data[indptr[i]:indptr[i+1]]``.
+
+        Args:
+            data (ti.ndarray): CSR format data array of the matrix.
+            indices (ti.ndarray): CSR format index array of the matrix.
+            indptr (ti.ndarray): CSR format index pointer array of the matrix.
+        """
+        if isinstance(data, Ndarray) and isinstance(
+                indices, Ndarray) and isinstance(indptr, Ndarray):
+            get_runtime().prog.make_sparse_matrix_from_ndarray_cusparse(
+                self.matrix, indptr.arr, indices.arr, data.arr)
+        else:
+            raise TaichiRuntimeError(
+                'Sparse matrix only supports building from [ti.ndarray, ti.Vectorndarray, ti.Matrix.ndarray]'
+            )
```

Comment:
It seems that, we only support `float32` as the value type. I think it is better to  check the type of data and prompt user a hint of supported type for now. :-)